jdbi/spotbugs-exclude.xml

```diff
@@ -15,17 +15,133 @@
     <Match>
         <!-- Exclude ANTLR generated files -->
         <Class name="~org\.skife\.jdbi\.rewriter\..*"/>
     </Match>
     <Match>
-        <!-- Legacy code -->
-        <Bug pattern="EI_EXPOSE_REP"/>
+        <Field type="java.lang.Class" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
     </Match>
     <Match>
-        <!-- Legacy code -->
-        <Bug pattern="EI_EXPOSE_REP2"/>
+        <Field type="java.lang.reflect.Method" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <Field type="java.sql.Connection" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <Field type="java.sql.PreparedStatement" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <Field type="java.sql.ResultSet" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <Field type="javax.sql.DataSource" />
+        <Bug pattern="EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <Field type="org.skife.jdbi.v2.StatementContext" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+
+    <!-- Most Joda time classes are immutable -->
+    <Match>
+        <Field type="org.joda.time.DateTime" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <!-- Most Joda time classes are immutable -->
+        <Field type="org.joda.time.DateTimeZone" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+
+    <Match>
+        <Field type="org.stringtemplate.v4.STGroup" />
+        <Bug pattern="EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <Field type="org.killbill.commons.metrics.api.MetricRegistry" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+    <Match>
+        <Field type="org.killbill.commons.embeddeddb.EmbeddedDB" />
+        <Bug pattern="EI_EXPOSE_REP2" />
+    </Match>
+
+    <Match>
+        <!--
+        Supposed to be mutable class that shared its state with other jdbi classes.
+        To test this, create copy constructor that copy its attributes, running test with "mvn -Pjdbi test verify" will fail.
+        -->
+        <Field type="org.skife.jdbi.v2.Binding" />
+        <Bug pattern="EI_EXPOSE_REP,EI_EXPOSE_REP2" />
+    </Match>
+
+    <Match>
+        <Class name="org.skife.jdbi.v2.ConcreteStatementContext" />
+        <!--
+        Try simple fix by wrapping with new Arraylist<>(cleanables) result with a lot of:
+        ".... org.skife.jdbi.v2.TestIterator.<xxx> .... java.lang.AssertionError: Handle was not closed correctly!"
+        when running test with: "mvn -Pjdbi test verify".
+        -->
+        <Method name="getCleanables" />
+        <Bug pattern="EI_EXPOSE_REP" />
+    </Match>
+
+    <!-- FIXME-COMMONS-113 - Is it worth investigating? as:
```

Comment:
Probably not 😄 

It was deprecated a long time ago (https://github.com/jdbi/jdbi/commit/1390cfae67dec84f542241f89a07830697bd5b52) and is even still beta on jdbi3 (https://github.com/jdbi/jdbi3-oracle12/blob/0a9ed5c63787f3a8da7ce46fe9323c85d839d7e4/src/main/java/org/jdbi/v3/oracle12/OracleReturning.java#L50).

Regardless, there is much more work needed for us to support Oracle... You can delete the class for now.